xwiki-platform-core/xwiki-platform-menu/xwiki-platform-menu-ui/src/main/resources/Menu/UIExtensionSheet.xml

```diff
@@ -56,12 +56,16 @@
   ## Prepare the extension point map.
   #set ($menuPanel = "{{velocity}}
     ${escapetool.h}panelheader(""$doc.plainTitle"")
-    {{menu type=""vertical collapsible open""}}{{include reference=""${escapetool.d}uix.doc.documentReference"" /}}{{/menu}}
+    {{menu type=""vertical collapsible open""
+    label=""$doc.plainTitle""}}{{include
+    reference=""${escapetool.d}uix.doc.documentReference""
+    /}}{{/menu}}
     ${escapetool.h}panelfooter()
     {{/velocity}}")
   #set ($menuBar = "{{velocity}}
     ${escapetool.h}if (${escapetool.d}xwiki.hasAccessLevel('view', ${escapetool.d}xcontext.user, ${escapetool.d}uix.doc.documentReference))
-      {{menu type=""horizontal fixedWidth"" id=""menu-horizontal-$doc.fullName.replaceAll('[\s|,|.|:|\[|\]|\[\/\]|\[\\\]|=|@|#]', '-')""}}{{include reference=""${escapetool.d}uix.doc.documentReference"" /}}{{/menu}}
+      {{menu type=""horizontal fixedWidth""
+      id=""menu-horizontal-$doc.fullName.replaceAll('[\s|,|.|:|\[|\]|\[\/\]|\[\\\]|=|@|#]', '-')"" label=""$doc.plainTitle""}}{{include reference=""${escapetool.d}uix.doc.documentReference"" /}}{{/menu}}
```

Comment:
You shouldn't use `$doc.plainTitle` without escaping, this can contain arbitrary text. Also this `fullName.replaceAll` looks quite dangerous to miss stuff.

Reply:
Yeah, as I've said, it's probably more or less fine, and in particular fine from a security point of view, but when you add escaping to the new stuff, I think it would be nice if you could escape this (for XWiki syntax), too.

Reply:
Actually, I think I forgot to submit one comment, so what I wanted to add is that the id attribute in HTML5 allows basically everything as long as it is non-empty and no ascii whitespace, so the important part is to ensure it is properly escaped. Citing [the specification](https://html.spec.whatwg.org/multipage/dom.html#the-id-attribute):

> When specified on [HTML elements](https://html.spec.whatwg.org/multipage/infrastructure.html#html-elements), the [id](https://html.spec.whatwg.org/multipage/dom.html#the-id-attribute) attribute value must be unique amongst all the [IDs](https://dom.spec.whatwg.org/#concept-id) in the element's [tree](https://dom.spec.whatwg.org/#concept-tree) and must contain at least one character. The value must not contain any [ASCII whitespace](https://infra.spec.whatwg.org/#ascii-whitespace).